scene/main/window.cpp

```diff
@@ -63,6 +63,17 @@ String Window::get_title() const {
 	return title;
 }
 
+void Window::set_icon(const Ref<Image> &p_icon) {
+	icon = p_icon;
+	if (icon.is_valid()) {
```

Comment:
```suggestion
	if (icon.is_valid()) {
```

Since it's not a node property, it should be possible to unset the icon by changing it to the `null`, so this check should not be checked here.